### ISSUE TITLE
NVDA doesn't read automatically the content of some types of dialog boxes

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#5447]
**Reported by k_kolev1985 on 2015-10-30 09:49**
It seams that NVDA does not read automatically (on focus) the content of some dialog boxes, related to some installers. If I'm not mistaking, those installers are from Microsoft (MSI) or from InstallShield (or maybe both). When such a dialog box appears, NVDA reads only its title and the currently focused button, but it doesn't read the main content of the dialog box. If the "read current window" command (NVDA+B) is pressed, NVDA reads everything in the dialog box, including its main content. One such dialog box in those installers is the one that prompts us to restart the system when (if) required. A good example for such an installer with such a dialog box is the one for "VMWare Tools" from the "VMWare Player" for virtual machines. The installer is used to install the required VMWare components in the created virtual machine itself and it is executed in the virtual machine itself. There are probably other such installers with such dialog boxes, but I can't remember specific examples at the moment.

Steps to reproduce it:
1. Run an installer created with MSI and one created with InstallShield. Make sure that it is for an application, the installation of witch will require a restart in the end, so the prompt for restart dialog box will appear. As I mentioned above, the one for VMWare Tools for a virtual machine in VMWare Player is a good example.
2. Complete the installation, by pressing "Finish" on the last screen of the installation process. A dialog box will appear, prompting you to restart the system in order to complete the installation.
3. Listen how NVDA will read the prompt for a restart dialog box, when it appears on the screen.
4. Then execute the "read current window" command (NVDA+B).
5. Listen now how NVDA will read the dialog box.

Actual results:
When the dialog box with the prompt for a restart appears, NVDA will read only its title and the currently focused button, but not the main dialog content text. When the "read current window" command (NVDA+B) is executed, NVDA will read the whole content of the dialog box (including the main content text), as it should.

Expected results:
When the dialog box with the prompt for a restart appears, NVDA should read the whole content of the dialog box, as it does when the "read current window" command (NVDA+B) is executed.

Test environment:
- Operating system: Windows 10 Pro (build 10240), 64-bit, in Bulgarian with all locale settings set to "Bulgarian", But it occurs on older versions of the OS as well.
- NVDA version: next-12679,8562dd5, but it occurs with older versions of NVDA as well.
- Processor: Intel Core i5-2320 at 3.00GHz
- RAM Memory: 4.00 GB.
- Graphics: Palit GeForce GT 630, 1024MB dedicated memory + 2047MB shared system memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC662 at Intel Cougar Point PCH - High Definition Audio Controller.

**Comment by bhavyashah** [comment]
Doesn't screen review or object navigation reveal the desired information from the dialog? Also, does this bug still exist?
P.S. Please mention the original reporter in case his username is known.

**Comment by k-kolev1985** [comment]
@bhavyashah Here are my answers to your questions:
1. I'm the reporter of this bug.
2. Yes, as far as I know (I'll test it to be certain later today|) - the bug still exists.
3. I guess that screen review or object navigation can read this information. But that is not the point here, because I can do it with NVDA+B as well on demand. The point is that NVDA should do it automatically when the dialog appears, but it does not do it.

**Comment by Brian1Gaff** [comment]
There are many times this does not seem to work. the classic one was always 
live mail but I've seen it in other programs as well. I think what is 
obviously not happening is that the software concerned does not actually 
flag to nvda that its new content and where it actually is, when you do 
nvda/b you read the whole lot  with no care about boundaries etc, indeed in 
some circumstances you can get continuous reading as if it goes back to the 
top and starts over for ever.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by Adriani90** [comment]
@k-kolev1985, @Brian1Gaff can you still reproduce this issue with NVDA 2018.4.1 or with the last Alpha?

**Comment by Adriani90** [comment]
@k-kolev1985 did this meanwhile somehow improve with NVDA last alpha version? Just to be sure whether we should still track this issue or not.

**Comment by k-kolev1985** [comment]
@Adriani90 Cannot check at the moment, since I don't have a case with such an installer with which to check.

**Comment by Adriani90** [comment]
Given updates are missing since many years, I am closing as abandoned. @k-kolev1985 in case you find an example installer where this issue is reproducible, please comment and we can reopen.
